src/main/java/nc/students/ayaz/repositories/UserRepository.java

```diff
@@ -0,0 +1,26 @@
+package nc.students.ayaz.repositories;
+
+import nc.students.ayaz.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+public class UserRepository {
+
+    private Map<String, User> users = new HashMap<>();
```

Comment:
&#x1F34F; **Resolved**
Agree. I just need to override hashCode in User class in order to use it in HashSet. And using HashSet becomes equal to use HashMap.